File: backend/dbust-backend/src/main/kotlin/com/fakebusters/dbust_backend/service/CSVService.kt
package com.fakebusters.dbust_backend.service

import org.springframework.stereotype.Service
import java.io.File
import java.io.FileWriter
import java.io.PrintWriter

@Service
class CSVService {

    private val csvFilePath = "upload_metrics.csv"

    init {
        // Create the CSV file and write the header if it doesn't exist
        val file = File(csvFilePath)
        if (!file.exists()) {
            PrintWriter(FileWriter(file, true)).use { writer ->
                writer.println("file_name,file_type,file_size(B),upload_time(ms)")
            }
        }
    }

    fun logUploadMetrics(fileName: String, fileType: String, fileSize: Long, uploadTime: Long) {
        val file = File(csvFilePath)
        PrintWriter(FileWriter(file, true)).use { writer ->
            writer.println("$fileName,$fileType,$fileSize,$uploadTime")
        }
    }
}

File: backend/dbust-backend-fastapi/csv_service.py
import os
import csv
from config import CSV_FILE_PATH

# Ensure CSV file exists and has the correct header
if not os.path.exists(CSV_FILE_PATH):
    with open(CSV_FILE_PATH, mode='w', newline='') as file:
        writer = csv.writer(file)
        writer.writerow(["file_name", "file_type", "file_size(B)", "upload_time(ms)"])

def log_upload_metrics(file_name, file_type, file_size, upload_time):
    with open(CSV_FILE_PATH, mode='a', newline='') as file:
        writer = csv.writer(file)
        writer.writerow([file_name, file_type, file_size, upload_time])